src/components/molecules/Schedule/Schedule.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useCallback, useMemo } from "react";
+import classNames from "classnames";
+import { format, fromUnixTime, getUnixTime } from "date-fns";
+import { range } from "lodash";
+import { useCss } from "react-use";
+
+import { RoomWithEvents, ScheduleProps } from "./Schedule.types";
+
+import { calcStartPosition } from "./Schedule.utils";
+
+import { ScheduleRoomEvents } from "../ScheduleRoomEvents";
+
+import "./Schedule.scss";
+
+const MAX_SCHEDULE_START_HOUR = 16;
+const MAX_HOUR = 24;
+const MIDDAY_HOUR = 12;
+
+export const Schedule: React.FC<ScheduleProps> = ({ scheduleDay }) => {
+  const getStartHour = useCallback(
+    (utcSeconds: number) => {
+      return utcSeconds >= scheduleDay.dayStartUtcSeconds
+        ? Number(format(fromUnixTime(utcSeconds), "H"))
+        : 0;
+    },
+    [scheduleDay]
+  );
+
+  const roomStartHour = useCallback(
+    (room: RoomWithEvents) =>
+      room.events?.reduce(
+        (acc, event) => Math.min(acc, getStartHour(event.start_utc_seconds)),
+        MAX_SCHEDULE_START_HOUR
+      ) ?? MAX_SCHEDULE_START_HOUR,
+    [getStartHour]
+  );
+
+  const scheduleStartHour = useMemo(
+    () =>
+      Math.min(
+        ...scheduleDay.rooms.map((room) => roomStartHour(room)),
+        MAX_SCHEDULE_START_HOUR
+      ),
+    [scheduleDay, roomStartHour]
+  );
+
+  const hours = useMemo(
+    () =>
+      range(scheduleStartHour, MAX_HOUR).map(
+        (hour: number) =>
+          `${hour % MIDDAY_HOUR || MIDDAY_HOUR} ${
+            hour >= MIDDAY_HOUR ? "PM" : "AM"
+          }`
+      ),
+    [scheduleStartHour]
+  );
+
+  const containerVars = useCss({
+    "--room-count": scheduleDay.rooms.length + 1,
```

Comment:
@mike-lvov What are you suggesting we move to the const `roomCount`? The `--room-count` itself, or the value?

Reply:
created variable for `1` (`NUMBER_OF_PERSONAL_ROOMS`). It's needed for the 1st line of the schedule (personalized). Which is not in the rooms list.
https://github.com/sparkletown/sparkle/pull/1302/commits/3773b544eaa9b24083615b54b444167ca724eff2

![image](https://user-images.githubusercontent.com/63313732/118135995-a011f400-b3d1-11eb-8e0e-38b42ae79eca.png)

Reply:
moved `NUMBER_OF_PERSONAL_ROOMS` to settings: https://github.com/sparkletown/sparkle/pull/1302/commits/e5721d0ad131b92d8d01a936e71f0f3a351d9ef1

Reply:
> It's needed for the 1st line of the schedule (personalized). Which is not in the rooms list.

Makes sense. Not sure that extracting it out into `src/settings` makes sense for this though. It seems to be heavily an implementation detail.

It would probably be sufficient to just have a comment at the point where you originally had the `+ 1` explaining its purpose there.

Reply:
done here: https://github.com/sparkletown/sparkle/pull/1302/commits/84c633a2848faab78da90654662b3d1b8aba1942